gemrb/core/Scriptable/Actor.cpp

```diff
@@ -3158,6 +3161,16 @@ void Actor::RefreshEffects(EffectQueue *fx)
 		UpdateAnimationID(true);
 	}
 
+	//delayed HP adjustment hack (after max HP modification)
+	//as it's triggered by PCFs from the previous tick, it should probably run before current PCFs
+	if (first && checkHP == 2) {
+		//could not set this in the constructor
+		checkHPTime = core->GetGame()->GameTime;
+	} else if (checkHP && checkHPTime != core->GetGame()->GameTime) {
+		checkHP = 0;
+		pcf_hitpoint(this, 0, BaseStats[IE_HITPOINTS]);
+	}
```

Comment:
Okay, how about here? Seems more natural, although I had to add a time check as it's reached way too often.

Reply:
Better, yes — thanks! It would probably also work by checking against Ticks, since that's saved to disk only for areas  — which don't have hp.